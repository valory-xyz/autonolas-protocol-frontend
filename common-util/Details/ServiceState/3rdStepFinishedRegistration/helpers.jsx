/* eslint-disable consistent-return */
export const isHashApproved = (contract, startingBlock, op) => new Promise((resolve, reject) => {
  const interval = setInterval(async () => {
    window.console.log('Attempting to getPastEvents...');

    try {
      const pastEvents = await contract.getPastEvents('ApproveHash', {
        filter: op,
        fromBlock: startingBlock - 10,
        toBlock: 'latest',
      });
      window.console.log('pastEvents:', pastEvents);

      const hashApproved = pastEvents.length !== 0;
      if (hashApproved) {
        window.console.log('hashApproved');
        clearInterval(interval);
        return resolve();
      }
    } catch (error) {
      clearInterval(interval);
      return reject(error);
    }
  }, 5000);
});
