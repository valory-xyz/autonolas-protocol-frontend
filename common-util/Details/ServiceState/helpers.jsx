export const SERVICE_STATE_HELPER_LABELS = {
  'pre-registration': 'The service has just been registered.',
  'active-registration':
    'The service is waiting for agent operators to register their agent instances.',
  'finished-registration':
    'All agent instance slots have been filled. Waiting for the service owner to continue deploying the service.',
  deployed:
    'The service is in default operational state. Agent operators can turn on their agent instances at this point.',
  terminated:
    'The service has been terminated by the service owner and the service is waiting for the owner to unbond all registered agents.',
};
