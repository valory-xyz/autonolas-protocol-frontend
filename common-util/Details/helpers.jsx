/* eslint-disable react/prop-types */
import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import {
  Button, Typography, Alert, Switch,
} from 'antd/lib';
import { ArrowUpRight, Circle } from 'react-feather';
import get from 'lodash/get';
import {
  DEFAULT_SERVICE_CREATION_ETH_TOKEN_ZEROS,
  GATEWAY_URL,
  NA,
  NAV_TYPES,
} from 'util/constants';
import { isL1OnlyNetwork } from 'common-util/functions';
import { NftImage } from './NFTImage';
import { SetOperatorStatus, OperatorWhitelist } from './ServiceDetailsHelper';
import {
  getTokenDetailsRequest,
  setOperatorsCheckRequest,
  checkIfServiceRequiresWhiltelisting,
} from './ServiceState/utils';
import {
  SubTitle,
  Info,
  SectionContainer,
  EachSection,
  ServiceStatus,
} from './styles';

const { Link, Text } = Typography;

const pattern = /https:\/\/localhost\/(agent|component|service)\/+/g;

export const getAutonolasTokenUri = (tokenUri) => (tokenUri || '').replace(pattern, GATEWAY_URL);

export const HASH_DETAILS_STATE = {
  IS_LOADING: 'IS_LOADING',
  LOADED: 'LOADED',
  FAILED: 'FAILED',
};

export const DetailsInfo = ({
  id,
  isOwner,
  type,
  tokenUri,
  info,
  metadata,
  metadataState,
  detailsOwner,
  onUpdateHash,
  setIsModalVisible,
  onDependencyClick,
}) => {
  const chainId = useSelector((state) => state?.setup?.chainId);

  const account = useSelector((state) => state?.setup?.account);
  const [tokenAddress, setTokenAddress] = useState(null);

  // switch state
  const [isWhiteListed, setIsWhiteListed] = useState(false);
  const [switchValue, setSwitchValue] = useState(isWhiteListed);
  useEffect(() => {
    setSwitchValue(isWhiteListed);
  }, [isWhiteListed]);

  // get token address for service on load
  useEffect(() => {
    const getData = async () => {
      if (type === NAV_TYPES.SERVICE) {
        const response = await getTokenDetailsRequest(id);
        setTokenAddress(response.token);
      }
    };

    if (id && isL1OnlyNetwork(chainId)) getData();
  }, [id]);

  const updateHashBtn = isOwner ? (
    <>
      {onUpdateHash && (
        <Button type="primary" ghost onClick={() => setIsModalVisible(true)}>
          Update Hash
        </Button>
      )}
    </>
  ) : null;

  const viewHashAndCode = HASH_DETAILS_STATE.LOADED === metadataState ? (
    <>
      {type === NAV_TYPES.SERVICE && <>&nbsp;•&nbsp;</>}
      <Link
        target="_blank"
        data-testid="view-hash-link"
        href={getAutonolasTokenUri(tokenUri)}
      >
        View Hash&nbsp;
        <ArrowUpRight size={16} />
      </Link>
        &nbsp;•&nbsp;
      <Link
        target="_blank"
        data-testid="view-code-link"
        href={(get(metadata, 'code_uri') || '').replace(
          'ipfs://',
          GATEWAY_URL,
        )}
      >
        View Code&nbsp;
        <ArrowUpRight size={16} />
      </Link>
    </>
  ) : null;

  // get operator whitelist
  const setOpWhitelist = async () => {
    if (isL1OnlyNetwork(chainId)) {
      const whiteListRes = await checkIfServiceRequiresWhiltelisting(id);
      setIsWhiteListed(whiteListRes);
    }
  };

  const getCommonDetails = () => {
    const commonDetails = [];

    if (HASH_DETAILS_STATE.LOADED === metadataState) {
      commonDetails.push(
        {
          title: 'Description',
          dataTestId: 'description',
          value: get(metadata, 'description') || NA,
        },
        {
          title: 'Version',
          dataTestId: 'version',
          value: get(metadata, 'attributes[0].value') || NA,
        },
      );
    }

    // If metadata failed, that means it has been unpinned from IPFS
    // and show an alert indicating the user
    if (HASH_DETAILS_STATE.FAILED === metadataState) {
      commonDetails.push({
        dataTestId: 'metadata-failed-to-load',
        value: (
          <Alert
            message="Metadata is unpinned from IPFS server"
            type="warning"
            showIcon
          />
        ),
      });
    }

    commonDetails.push({
      title: 'Owner Address',
      dataTestId: 'owner-address',
      value: detailsOwner || NA,
    });

    return commonDetails;
  };

  const getComponentAndAgentValues = () => {
    const dependencies = get(info, 'dependencies') || [];
    return [
      {
        dataTestId: 'hashes-list',
        value: (
          <>
            {viewHashAndCode}
            {updateHashBtn}
          </>
        ),
      },
      ...getCommonDetails(),
      {
        title: 'Component Dependencies',
        dataTestId: 'details-dependency',
        value:
          dependencies.length === 0 ? (
            <>None</>
          ) : (
            dependencies.map((e) => (
              <li key={`${type}-dependency-${e}`}>
                <Button type="link" onClick={() => onDependencyClick(e)}>
                  {e}
                </Button>
              </li>
            ))
          ),
      },
    ];
  };

  const getServiceValues = () => {
    const serviceState = ['2', '3', '4'].includes(get(info, 'state'));
    const serviceDetailsList = [
      {
        dataTestId: 'hashes-list',
        value: (
          <>
            <ServiceStatus
              className={serviceState ? 'active' : 'inactive'}
              data-testid="service-status"
            >
              <Circle size={8} />
              <Text>{serviceState ? 'Active' : 'Inactive'}</Text>
            </ServiceStatus>
            {viewHashAndCode}
          </>
        ),
      },
    ];

    // show NFT image only if metadata is available
    if (HASH_DETAILS_STATE.LOADED === metadataState) {
      serviceDetailsList.push({
        dataTestId: 'service-nft-image',
        value: <NftImage metadata={metadata} type={type} />,
      });
    }

    serviceDetailsList.push(...getCommonDetails(), {
      title: 'Threshold',
      value: get(info, 'threshold', null) || NA,
    });

    console.log({
      chainId,
      tokenAddress,
      isNotEq: tokenAddress !== DEFAULT_SERVICE_CREATION_ETH_TOKEN_ZEROS,
    });

    // show token address only if it is not ETH
    if (
      isL1OnlyNetwork(chainId)
      && tokenAddress !== DEFAULT_SERVICE_CREATION_ETH_TOKEN_ZEROS
    ) {
      serviceDetailsList.push({
        title: 'Token Address',
        value: tokenAddress || NA,
      });
    }

    // operator whitelisting is only available for service
    serviceDetailsList.push({
      title: (
        <>
          Operator Whitelisting&nbsp;
          <Switch
            disabled={!isOwner}
            checked={switchValue}
            checkedChildren="Enabled"
            unCheckedChildren="Disabled"
            onChange={async (checked) => {
              setSwitchValue(checked);
              if (!checked) {
                await setOperatorsCheckRequest({
                  account,
                  serviceId: id,
                  isChecked: false,
                });
                await setOpWhitelist();
              }
            }}
          />
        </>
      ),
      value: (
        <OperatorWhitelist
          id={id}
          setOpWhitelist={setOpWhitelist}
          isWhiteListed={isWhiteListed}
        />
      ),
    });

    if (isOwner) {
      serviceDetailsList.push({
        title: 'Set operators statuses',
        value: <SetOperatorStatus id={id} />,
      });
    }

    return serviceDetailsList;
  };

  const details = type === NAV_TYPES.SERVICE
    ? getServiceValues()
    : getComponentAndAgentValues();

  return (
    <SectionContainer>
      {details.map(({ title, value, dataTestId }, index) => (
        <EachSection key={`${type}-details-${index}`}>
          {title && <SubTitle strong>{title}</SubTitle>}
          <Info data-testid={dataTestId || ''}>{value}</Info>
        </EachSection>
      ))}
    </SectionContainer>
  );
};
