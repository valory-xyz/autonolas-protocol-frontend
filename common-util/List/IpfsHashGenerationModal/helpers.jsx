/* eslint-disable import/no-unresolved */
/* eslint-disable no-restricted-syntax */

import { create } from 'ipfs-http-client';
import { base16 } from 'multiformats/bases/base16';
import { HASH_PREFIXES } from 'util/constants';

const ipfs = create({
  host: process.env.NEXT_PUBLIC_REGISTRY_URL,
  port: 443,
  protocol: 'https',
});

export const getIpfsHashHelper = async (info, hashType) => {
  const updatedInfo = {
    ...info,
    code_uri: `ipfs://${hashType}${info.code_uri}`,
    attributes: [
      {
        trait_type: 'version',
        value: info.version,
      },
    ],
  };

  /**
   * metadata should be compatible with standards
   * https://docs.opensea.io/docs/metadata-standards
   */
  delete updatedInfo.version;

  const otherOptions = {
    wrapWithDirectory: false, // default: false
  };

  const response = await ipfs.add(
    { path: 'metadata.json', content: JSON.stringify(updatedInfo) },
    otherOptions,
  );

  const hash = response.cid.toV1().toString(base16.encoder);

  // the response will always be of the 'type 1' so replace it.
  const updatedHash = hash.replace(HASH_PREFIXES.type1, '');
  return updatedHash;
};
