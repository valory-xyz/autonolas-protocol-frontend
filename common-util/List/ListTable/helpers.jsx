import { useState } from 'react';
import {
  Input, Space, Button, Typography, Tooltip,
} from 'antd/lib';
import { SearchOutlined, CopyOutlined } from '@ant-design/icons';
import PropTypes from 'prop-types';

import { NAV_TYPES, SERVICE_STATE, TOTAL_VIEW_COUNT } from 'util/constants';

const { Text, Title } = Typography;
const textStyle = { maxWidth: '100%' };

/**
 * helper components
 */

export const getTrimmedText = (str, suffixCount) => {
  const text = str.trim();
  const frontText = text.slice(0, suffixCount);
  const backText = text.slice(text.length - suffixCount, text.length);
  return `${frontText}...${backText}`;
};

export const EllipsisMiddle = ({ suffixCount, children, ...rest }) => {
  if (typeof children !== 'string') return <>{children}</>;

  if (children.length <= 12) return <Text {...rest}>{children}</Text>;

  /**
   * truncate only if the character exceeds more than 12
   */
  return (
    <Text style={textStyle} {...rest}>
      {getTrimmedText(children, suffixCount)}
      <Tooltip title="Copy">
        &nbsp;
        <Button
          onClick={() => navigator.clipboard.writeText(children)}
          icon={<CopyOutlined />}
        />
      </Tooltip>
    </Text>
  );
};

EllipsisMiddle.propTypes = {
  suffixCount: PropTypes.number,
  children: PropTypes.string,
};

EllipsisMiddle.defaultProps = {
  suffixCount: 6,
  children: '',
};

/**
 * helper functions
 */

export const getTableColumns = (type, { onViewClick, onUpdateClick }) => {
  if (type === NAV_TYPES.COMPONENT || type === NAV_TYPES.AGENT) {
    return [
      {
        title: 'ID',
        dataIndex: 'id',
        key: 'id',
        width: 50,
      },
      {
        title: 'Owner',
        dataIndex: 'owner',
        key: 'owner',
        width: 160,
        render: (text) => <EllipsisMiddle>{text}</EllipsisMiddle>,
      },
      {
        title: 'Hash',
        dataIndex: 'hash',
        key: 'hash',
        width: 200,
        render: (text) => <EllipsisMiddle>{text}</EllipsisMiddle>,
      },
      {
        title: 'No. of component dependencies',
        dataIndex: 'dependency',
        width: 180,
        key: 'dependency',
      },
      {
        width: 200,
        title: 'Action',
        key: 'action',
        fixed: 'right',
        render: (_text, record) => (
          <Space size="middle">
            <Button type="link" onClick={() => onViewClick(record.id)}>
              View
            </Button>
          </Space>
        ),
      },
    ];
  }

  if (type === NAV_TYPES.SERVICE) {
    return [
      {
        title: 'ID',
        dataIndex: 'id',
        key: 'id',
        width: 50,
        fixed: 'left',
      },
      {
        title: 'Owner',
        dataIndex: 'owner',
        key: 'owner',
        width: 200,
        render: (text) => <EllipsisMiddle>{text}</EllipsisMiddle>,
      },
      {
        title: 'State',
        dataIndex: 'state',
        key: 'state',
        width: 150,
        render: (e) => <>{SERVICE_STATE[e]}</>,
      },
      {
        width: 220,
        title: 'Action',
        key: 'action',
        fixed: 'right',
        render: (_text, record) => {
          const canUpdate = record.state === '3' || record.state === '4' || record.state === '5';

          return (
            <Space size="middle">
              <Button type="link" onClick={() => onViewClick(record.id)}>
                View
              </Button>

              {canUpdate && onUpdateClick && (
                <Button type="link" onClick={() => onUpdateClick(record.id)}>
                  Update
                </Button>
              )}
            </Space>
          );
        },
      },
    ];
  }

  return [];
};

export const getData = (type, rawData, { current }) => {
  /**
   * @example
   * TOTAL_VIEW_COUNT = 10, current = 1
   * start = ((1 - 1) * 10) + 1
   *       = (0 * 10) + 1
   *       = 1
   * TOTAL_VIEW_COUNT = 10, current = 1
   * start = ((5 - 1) * 10) + 1
   *       = 40 + 1
   *       = 41
   */
  const startIndex = (current - 1) * TOTAL_VIEW_COUNT + 1;
  let data = [];
  if (type === NAV_TYPES.COMPONENT) {
    data = rawData.map((item, index) => ({
      id: item.id || `${startIndex + index}`,
      description: item.description || '-',
      developer: item.developer || '-',
      owner: item.owner || '-',
      hash: item.unitHash || '-',
      dependency: (item.dependencies || []).length,
    }));
  }

  if (type === NAV_TYPES.AGENT) {
    data = rawData.map((item, index) => ({
      id: item.id || `${startIndex + index}`,
      description: item.description || '-',
      developer: item.developer || '-',
      owner: item.owner || '-',
      hash: item.unitHash || '-',
      dependency: (item.dependencies || []).length,
    }));
  }

  if (type === NAV_TYPES.SERVICE) {
    data = rawData.map((item, index) => ({
      id: item.id || `${startIndex + index}`,
      developer: item.developer || '-',
      owner: item.owner || '-',
      active: `${item.active}`,
      state: item.state,
    }));
  }

  return data;
};

/**
 * tab content
 */
export const useExtraTabContent = ({ title, onRegisterClick = () => {} }) => {
  const [searchValue, setSearchValue] = useState('');
  const [value, setValue] = useState('');
  const clearSearch = () => {
    setValue('');
    setSearchValue('');
  };

  const extraTabContent = {
    left: <Title level={2}>{title}</Title>,
    right: (
      <>
        <Input
          prefix={<SearchOutlined className="site-form-item-icon" />}
          placeholder="Owner or Hash"
          value={value}
          onChange={(e) => setValue(e.target.value)}
        />
        <Button
          ghost
          type="primary"
          onClick={() => setSearchValue(value || '')}
        >
          Search
        </Button>
        <Button type="primary" onClick={onRegisterClick}>
          Mint
        </Button>
      </>
    ),
  };

  return { searchValue, extraTabContent, clearSearch };
};

/**
 * returns hash from the url
 * @example
 * input: router-path (for example, /components#my-components)
 * output: my-components
 */
export const getHash = (router) => router?.asPath?.split('#')[1] || '';

/**
 * my-components/my-agents/my-serices has "my" in common hence returns
 */
export const isMyTab = (hash) => !!(hash || '').includes('my-');
