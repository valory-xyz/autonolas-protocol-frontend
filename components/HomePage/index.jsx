import { useSelector } from 'react-redux';
import {
  Button, Col, Row, Typography,
} from 'antd';
import Link from 'next/link';
import { URL } from 'util/constants';
import { isL1Network } from 'common-util/functions';
import { useScreen } from 'common-util/hooks/useScreen';
import { Container, HeaderRow, ContentRow } from './styles';

const { Title, Text } = Typography;

const IMG_PATH = 'images/homepage/';

const LIST = [
  {
    type: 'services',
    title: 'Service owners mint and manage complete agent services',
    desc: 'They manage how the service is set up. Service owners define the business model of their services, and charge DAOs to use them.',
    link: URL.SERVICES,
  },
  {
    type: 'agents',
    title: 'Services are made of agents',
    desc: 'Services are run by multiple  software agents, each with an independent operator. Developers can author full agents, and these can be composed by service owners.',
    link: URL.AGENTS,
  },
  {
    type: 'components',
    title: 'Each agent is made of components',
    desc: 'Components are smaller blocks of code which can be reused by agent developers.',
    link: URL.COMPONENTS,
  },
];

const HomePage = () => {
  const chainId = useSelector((state) => state?.setup?.chainId);
  const { isMobile } = useScreen();

  return (
    <Container>
      <HeaderRow className="row-1">
        <Row>
          <Col span={14} offset={2}>
            <Title className="hero-title">
              {` Mint and manage your services${
                isL1Network(chainId) ? ', agents and components' : ''
              }.`}
            </Title>
            <Text className="lead">
              The easiest way to interact with the Autonolas on-chain registry.
            </Text>
            <Link
              href={isL1Network(chainId) ? URL.COMPONENTS : URL.SERVICES}
              passHref
              legacyBehavior
            >
              <Button type="primary" size="large">
                Get started â†’
              </Button>
            </Link>
          </Col>
          <Col span={8}>
            <div
              className="header-image"
              style={{
                backgroundImage: `url(${IMG_PATH}autonomous-agent-service-architecture.svg)`,
              }}
            />
          </Col>
        </Row>
      </HeaderRow>

      <ContentRow className="row-2">
        <Title level={3} className="title">
          How are Autonolas services architected?
        </Title>

        {LIST.map(({
          title, desc, link, type,
        }) => (
          <Row
            className="each-service"
            key={`each-service-${type}`}
            align="middle"
          >
            <Col className="column column-1" md={12} xs={10}>
              <img
                src={`/${IMG_PATH}${type}.svg`}
                className="each-service-image"
                alt={desc}
                width={isMobile ? 120 : 150}
                height={isMobile ? 120 : 150}
              />
            </Col>

            <Col className="column column-2" md={12} xs={14}>
              <Title level={4}>{title}</Title>
              <Text className="description">{desc}</Text>
              <br />
              {type === 'services' ? (
                <Link href={link} legacyBehavior>{`View all ${type}`}</Link>
              ) : (
                <>
                  {isL1Network(chainId) ? (
                    <Link href={link} legacyBehavior>{`View all ${type}`}</Link>
                  ) : (
                    <Text disabled>
                      {`Switch network (to Ethereum or Goerli) to view ${type}`}
                    </Text>
                  )}
                </>
              )}
            </Col>
          </Row>
        ))}
      </ContentRow>
    </Container>
  );
};

export default HomePage;
