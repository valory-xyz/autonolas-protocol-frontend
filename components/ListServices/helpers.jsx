import compact from 'lodash/compact';

export const getNumberOfAgentAddress = (agentAddresses) => {
  /**
   * get the number of addresses
   * g1. ['0x123', '0x456'] => 2
   * eg2. ['0x123', '0x456', ''] => 2 // empty string (falsy) is ignored
   */
  const addressCount = compact((agentAddresses || '').split(',')).length;
  return addressCount;
};

/**
 * FOR AGENT ID
 * 1. get the bond value
 * 2. get the number of input addresses
 * 3. multiply the number of past addresses with the bond value
 *
 * @example
 * input: [agentId1 => 2 address, agentId2 => 3 address]
 * bonds: [100, 200]
 * output: 2 * 100 + 3 * 200 = 800
 */
export const transformSlotsAndBonds = (
  slotsArray,
  bondsArray,
  tableDataSource,
) => {
  let totalBonds = 0;
  (tableDataSource || []).forEach((data) => {
    const { agentAddresses, bond } = data;

    /**
     * get the number of addresses
     * g1. ['0x123', '0x456'] => 2
     * eg2. ['0x123', '0x456', ''] => 2 // empty string (falsy) is ignored
     */
    const numberOfAgentAddress = getNumberOfAgentAddress(agentAddresses);

    // multiply the number of addresses with the bond value of the agentId
    totalBonds += numberOfAgentAddress * bond;
  });

  return { slots: slotsArray, bonds: bondsArray, totalBonds };
};

export const transformDatasourceForServiceTable = ({
  agentIds,
  numAgentInstances,
  slots,
  bonds,
}) => {
  const dateSource = agentIds.map((aid, i) => ({
    key: aid,
    agentId: aid,
    availableSlots: Number(slots[i]) - Number(numAgentInstances[i]),
    totalSlots: slots[i],
    bond: bonds[i],
    agentAddresses: null,
  }));

  return dateSource;
};
