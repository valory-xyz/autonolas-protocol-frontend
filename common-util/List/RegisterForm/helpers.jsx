import PropTypes from 'prop-types';
import { Form, Input, Tooltip } from 'antd';
import { LinkOutlined } from '@ant-design/icons';
import { GATEWAY_URL, HASH_PREFIX } from 'util/constants';
import { getBase16Validator } from '../IpfsHashGenerationModal';

export const FormItemHash = ({ listType, hashValue }) => (
  <Form.Item
    label="Hash of Metadata File"
    name="hash"
    rules={[
      {
        required: true,
        message: `Please input the IPFS hash of the ${listType}`,
      },
      () => ({
        validator(_, value) {
          return getBase16Validator(value);
        },
      }),
    ]}
  >
    <Input
      disabled
      addonBefore={HASH_PREFIX}
      addonAfter={(
        <Tooltip
          title={
            hashValue
              ? 'Click to open the generated hash'
              : 'Please generate hash'
          }
        >
          <LinkOutlined
            style={hashValue ? {} : { cursor: 'not-allowed' }}
            onClick={() => {
              if (hashValue) {
                window.open(
                  `${GATEWAY_URL}${HASH_PREFIX}${hashValue}`,
                  '_blank',
                );
              }
            }}
          />
        </Tooltip>
      )}
    />
  </Form.Item>
);

FormItemHash.propTypes = {
  listType: PropTypes.string,
  hashValue: PropTypes.string,
};

FormItemHash.defaultProps = {
  listType: '',
  hashValue: null,
};

export const getX = () => {};
