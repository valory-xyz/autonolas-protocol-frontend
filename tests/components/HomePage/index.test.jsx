import React from 'react';
import { render } from '@testing-library/react';
import HomePage from 'components/HomePage';
import { useRouter } from 'next/router';
import { wrapProvider } from '../../helpers';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

useRouter.mockImplementation(() => ({ push: jest.fn() }));

// test cases
describe('<HomePage>', () => {
  it('should render homepage', async () => {
    expect.hasAssertions();
    const { container } = render(wrapProvider(<HomePage />));
    expect(container).toMatchSnapshot();
  });
});
