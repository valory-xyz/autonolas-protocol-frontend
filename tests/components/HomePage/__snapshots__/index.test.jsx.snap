// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage> should render homepage 1`] = `
<div>
  <div
    class="styles__Container-sc-4qo35f-0"
  >
    <div
      class="styles__HeaderRow-sc-4qo35f-1 cWafSz row-1"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-14 ant-col-offset-2"
        >
          <h1
            class="ant-typography hero-title"
          >
             Mint and manage your services, agents and components.
          </h1>
          <span
            class="ant-typography lead"
          >
            The easiest way to interact with the Autonolas on-chain registry.
          </span>
          <a
            class="ant-btn ant-btn-primary ant-btn-lg"
            href="/components"
          >
            <span>
              Get started â†’
            </span>
          </a>
        </div>
        <div
          class="ant-col ant-col-8"
        >
          <div
            class="header-image"
            style="background-image: url(images/homepage/autonomous-agent-service-architecture.svg);"
          />
        </div>
      </div>
    </div>
    <div
      class="styles__ContentRow-sc-4qo35f-2 bHSfXW row-2"
    >
      <h3
        class="ant-typography title"
      >
        How are Autonolas services architected?
      </h3>
      <div
        class="ant-row ant-row-middle each-service"
      >
        <div
          class="ant-col column column-1"
        >
          <img
            alt=""
            class="each-service-image"
            src="/images/homepage/services.svg"
          />
        </div>
        <div
          class="ant-col column column-2"
        >
          <h5
            class="ant-typography"
          >
            Service owners mint and manage complete agent services
          </h5>
          <span
            class="ant-typography description"
          >
            They manage how the service is set up. Service owners define the business model of their services, and charge DAOs to use them.
          </span>
          <br />
          <a
            href="/services"
          >
            View all services
          </a>
        </div>
      </div>
      <div
        class="ant-row ant-row-middle each-service"
      >
        <div
          class="ant-col column column-1"
        >
          <img
            alt=""
            class="each-service-image"
            src="/images/homepage/agents.svg"
          />
        </div>
        <div
          class="ant-col column column-2"
        >
          <h5
            class="ant-typography"
          >
            Services are made of agents
          </h5>
          <span
            class="ant-typography description"
          >
            Services are run by multiple  software agents, each with an independent operator. Developers can author full agents, and these can be composed by service owners.
          </span>
          <br />
          <a
            href="/agents"
          >
            View all agents
          </a>
        </div>
      </div>
      <div
        class="ant-row ant-row-middle each-service"
      >
        <div
          class="ant-col column column-1"
        >
          <img
            alt=""
            class="each-service-image"
            src="/images/homepage/components.svg"
          />
        </div>
        <div
          class="ant-col column column-2"
        >
          <h5
            class="ant-typography"
          >
            Each agent is made of components
          </h5>
          <span
            class="ant-typography description"
          >
            Components are smaller blocks of code which can be reused by agent developers.
          </span>
          <br />
          <a
            href="/components"
          >
            View all components
          </a>
        </div>
      </div>
    </div>
    <br />
    <br />
  </div>
</div>
`;
